pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -502,5 +514,23 @@ func (s *Reconciler) createVirtualMachine(ctx context.Context, nicName string) e
 		}
 	}
 
-	return err
+	return nil
+}
+
+func (s *Reconciler) getCustomUserData() (string, error) {
+	userData := ""
+	if s.scope.MachineConfig.UserDataSecret != nil {
```

Comment:
This is a nit for me in this PR now... but if you invert the test, the indentation reduces and the other `if...else` could be simpler too.

Reply:
I couldnt figure out a way to fixup previous commit, created a new one for simplicity, hope thats ok

Reply:
FYI you can rewrite and edit your e.g last 2 commits with `git rebase -i HEAD~2`
Additionally you might also find useful `git stash save/pop`